include/sacn/merge.h

```diff
@@ -0,0 +1,162 @@
+ * buffers and update two result buffers:
+ *   512 bytes for the merged data values (i.e. "winning level").  These are calculated by using
+ *     a Highest-Level-Takes-Precedence(HTP) algorithm for all sources that share the highest
+ *     per-address priority.
+ *   512 source identifiers (i.e. "winning source") to indicate which source was considered the
+ *     source of the merged data value, or that no source currently owns this address.
+ *
+ * \TODO: Add sample usage
+ *
+ * @{
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/*! Each universe has a handle associated with it.*/
+typedef int universe_handle_t;
+
+/*! The sources on a universe have a short id that is used in the owned values, rather than a UUID.*/
+typedef uint16_t source_id_t;
+
+/*! An invalid source id handle value. */
+#define DMX_MERGER_SOURCE_INVALID -1
+
+/*! A set of configuration information for a universe to be merged. */
+typedef struct DmxMergerUniverseConfig
+{
+  /*! The maximum number of sources this universe will listen to.  May be #SACN_RECEIVER_INFINITE_SOURCES.
+      This parameter is ignored when configured to use static memory -- #DMX_MERGER_MAX_SOURCES_PER_UNIVERSE is used
+      instead.*/
+  size_t source_count_max;
+
+  /*! Buffer of DMX_ADDRESS_COUNT levels that this library keeps up to date as it merges.
+      Memory is owned by the application.*/
+  uint8_t* slots;
+
+  /*! Buffer of DMX_ADDRESS_COUNT source IDs that indicate the current winner of the merge for
+      that slot, or DMX_MERGER_SOURCE_INVALID to indicate that no source is providing values for that slot.
+      Memory is owned by the application.*/
+  source_id_t* slot_owners;
+
+} DmxMergerUniverseConfig;
+
+/*!
+ * \brief An initializer for an DmxMergerUniverseConfig struct.
+ *
+ * Usage:
+ * \code
+ * // Create the struct
+ * DmxMergerUniverseConfig universe_config = DMX_MERGER_UNIVERSE_CONFIG_INIT;
+ * // Now fill in the members of the struct
+ * \endcode
+ *
+ */
+#define DMX_MERGER_UNIVERSE_CONFIG_INIT \
+  {                                     \
+    0, NULL, NULL                       \
+  }
+
+/*! The current input data for a single source of the merge.
+    This is exposed only for informational purposes, as the
+    application call a variant of *TODO* */
+typedef struct DmxMergerSource
+{
+  /*! The UUID (e.g. sACN CID) of the DMX source. */
+  EtcPalUuid cid;
+
+  /*! The DMX data values (0 - 255). */
+  uint8_t values[DMX_ADDRESS_COUNT];
+
+  /*! Some sources don't send all 512 values, so here's how much of values to use.*/
+  size_t valid_value_count;
+
+  /*! The sACN per-universe priority (0 - 255). */
+  uint8_t universe_priority;
+
+  /*! Whether or not the address_priority buffer is valid. */
+  bool address_priority_valid;
+
+  /*! The sACN per-address (startcode 0xdd) priority (1-255, 0 means not sourced).
+      If the source does not */
+  uint8_t address_priority[DMX_ADDRESS_COUNT];
+
+} DmxMergerSource;
+
+etcpal_error_t dmx_merger_init();
```

Comment:
I'm not a huge fan of the name `dmx_merger`. For a few reasons:

* It does not start with `sacn_` and so breaks the convention of the rest of this library and other libraries, that try to always pseudo-namespace the library name to the front of all modules.
* "merger" is a noun, but it doesn't describe the handle/object instance that is created when using this API. For a counterexample, `sacn_receiver_create()` creates an sACN receiver instance, which you access using a handle called `sacn_receiver_t`. But these functions create "universes", which you access using a `universe_handle_t`.

A few alternatives that come to mind:

**sacn_merger**
```c
/*! \file sacn/merger.h */

typedef int sacn_merger_t

etcpal_error_t sacn_merger_create(const SacnMergerConfig* config, sacn_merger_t* handle);
```

**sacn_dmx_merger**
```c
/*! \file sacn/dmx_merger.h */

typedef int sacn_dmx_merger_t;

etcpal_error_t sacn_dmx_merger_create(const SacnDmxMergerConfig* config, sacn_dmx_merger_t* handle);
```

Notice in all cases that I've tried to keep mnemonic consistency between the file name, the handle type, and the API function names as much as possible.

Reply:
The main reason I chose dmx_merger (and had it in its own library) was because I wanted to emphasize that this could be used for sACN or in combination with other DMX sources.   But I do see your point.  I'll play with your suggestions.

Reply:
Everything should be changed over to sacn_dmx_merger now.  This feels right -- I didn't really like the notion of universe bleeding into this module, and now I was able to remove it.  Let me know if you have other issues or resolve the conversation.